appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/annotation/impl/AppClientScanner.java

```diff
@@ -158,4 +167,34 @@ private URI scanURI(URI uriToAdd) throws IOException {
         }
         return uriToAdd;
     }
+
+    /**
+     * Overriding to handle the case where EJB class is mistakenly packages inside an appclient jar.
+     * Instead of throwing an error which might raise backward compatiability issues, a cleaner way is to
+     * just skip the annotation processing for them.
+     */
+    @Override
+    protected void calculateResults(ApplicationClientDescriptor bundleDesc) {
+        super.calculateResults(bundleDesc);
+
+        Class[] ejbAnnotations = ejbProvider.getAnnotationTypes();
```

Comment:
I didn't quite understand what they do in `getEjbAnnotationTypes` function call of `AbstractHandler`